R-package/tests/testthat/test_Predictor.R

```diff
@@ -111,3 +113,117 @@ test_that("start_iteration works correctly", {
     pred_leaf2 <- predict(bst, test$data, start_iteration = 0L, num_iteration = end_iter + 1L, predleaf = TRUE)
     expect_equal(pred_leaf1, pred_leaf2)
 })
+
+test_that("predictions keep row names from the data", {
+    data("mtcars")
+    X <- as.matrix(mtcars[, -1L])
+    y <- as.numeric(mtcars[, 1L])
+    dtrain <- lgb.Dataset(X, label = y, params = list(max_bins = 5L))
+    bst <- lgb.train(
+        data = dtrain
+        , obj = "regression"
+        , nrounds = 5L
+        , verbose = -1L
```

Comment:
```suggestion
        , verbose = VERBOSITY
```

Please use the constant defined at the top of this file to set verbosity.

https://github.com/microsoft/LightGBM/blob/17d4e007852b4f0b34b211a674cb26e6751c98e2/R-package/tests/testthat/test_Predictor.R#L1-L3

Per #4862, we're slowly working on reducing the logs produced by these tests, to make it easier to visually inspect test results. That code I've linked allows easily running the tests with different verbosity levels by passing an environment variables.